functionary/ui/views/environments/detail.py

```diff
@@ -0,0 +1,52 @@
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.views.generic.detail import DetailView
+
+from core.auth import Permission
+from core.models import Environment, Package, Variable
+
+from .utils import get_users
+
+
+class EnvironmentDetailView(LoginRequiredMixin, UserPassesTestMixin, DetailView):
+    model = Environment
+
+    def get_queryset(self):
+        return (
+            super()
+            .get_queryset()
+            .select_related(
+                "team",
+            )
+        )
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        env = self.get_object()
+        env_members = get_users(env)
+
+        context["packages"] = Package.objects.filter(environment=env)
+        context["environment_create_perm"] = (
+            True if (self.request.user.has_perm(Permission.TEAM_UPDATE, env)) else False
```

Comment:
I'm not sure if this is the right permission.

Reply:
Fixed